src/constructs/populateProvidedVpc/populateProvidedVpc.ts

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { Fn, Stack, Tags } from 'aws-cdk-lib';
+import { CfnSubnet, CfnSubnetRouteTableAssociation } from 'aws-cdk-lib/aws-ec2';
+import { Construct } from 'constructs';
+
+/**
+ * The tag value for `aws-cdk:subnet-type` which will be applied to a subnet
+ */
+export enum SubnetTag {
+  /**
+   * {"Name": "aws-cdk:subnet-type", "Value": "Private"} tag on a subnet
+   */
+  PRIVATE = 'Private',
+  /**
+   * {"Name": "aws-cdk:subnet-type", "Value": "Public"} tag on a subnet
+   */
+  PUBLIC = 'Public',
+  /**
+   * {"Name": "aws-cdk:subnet-type", "Value": "Isolated"} tag on a subnet
+   */
+  ISOLATED = 'Isolated',
+}
```

Comment:
CDK v2 uses [different enums](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.SubnetType.html) for SubnetTypes. Recommend:
* Verifying what tags the new enums create in VPCs.
* Using the same enums here.